searchlib/src/vespa/searchlib/queryeval/flow.h

```diff
@@ -285,6 +285,50 @@ auto select_forced_strict_and_child_with_order(auto adapter, const auto &childre
     return std::make_tuple(best_idx, best_target, best_diff);
 }
 
+auto select_forced_strict_and_child_with_destructive_order(auto adapter, const auto &children, size_t first) {
```

Comment:
Perhaps this can share implementation with `select_forced_strict_and_child_with_order()`, as the only difference is whether you can "_move past someone who will lose strictness_"?